### PR TITLE
Bump `flutter_template_images` to remove TODO.

### PR DESCRIPTION
**Pull request by matanlurey**
Towards https://github.com/flutter/flutter/issues/160692.

Prior to `flutter_template_images: 4.3.0`, the `app_shared` directory was required.

In 4.3.0 it was forked back out to `app` (https://pub.dev/packages/flutter_template_images/changelog), as the `skeleton` template has been removed, and the `skeleton` template was the (only) reason for `app_shared` (to share code between `app` and `skeleton`).

This is a NOP change tested by existing tests.

### CODE REVIEW

**Review by andrewkolos — APPROVED** [review]
LGTM

**Comment by yjbanov** [comment]
test-exempt: dead code removal

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ dependencies:
   crypto: 3.0.6
   ffi: 2.1.3
   file: 7.0.1
-  flutter_template_images: 4.2.0
+  flutter_template_images: 4.3.0
```
Comment (goderbauer) [inline packages/flutter_tools/pubspec.yaml:23]: I believe you'll have to run `flutter update-packages --force-upgrade` to recompute this file's checksum in line 124. I'd run it with the `--cherry-pick-package/version` option to only bump this package and not everything else.

Comment (matanlurey) [reply packages/flutter_tools/pubspec.yaml:23]: Looks like you can't mix --cherry-pick _and_ --force-upgrade, but just --chery-pick-xxx worked.

**Pull request merged**